frontend/www/js/omegaup/components/course/StudentProgress.vue

```diff
@@ -120,6 +121,27 @@ export default class StudentProgress extends Vue {
     );
   }
 
+  get weightedMean(): Array<number> {
+    const totalPoints =
+      this.assignments
+        .map((assignment) => assignment.max_points)
+        .reduce((a, b) => (a ?? 0) + (b ?? 0)) ?? 0;
+    return this.assignments.map(
+      (assignment) => ((assignment.max_points ?? 0) / (totalPoints ?? 0)) * 100,
+    );
+  }
+
+  get globalScoreByStudent(): string {
+    const points = [];
+    for (const assignment of this.assignments) {
+      points.push(this.score(assignment.alias) / this.points(assignment.alias));
+    }
```

Comment:
```suggestion
    const points = this.assignments.map((assignment) => 
      this.score(assignment.alias) / this.points(assignment.alias));
```

esto es un poco más conciso.

pregunta: `this.points()` puede regresar 0?

Reply:
No estoy completamente seguro (por los contenidos que sólo incluyen lecturas), pero para evitar división entre 0, agregaré una condición.